### PR TITLE
migrate to Xcode 10.2 and Swift 5

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/392891325557410/1116046264056095
Tech Design URL:
CC:

**Description**:

Updates dependencies to those that compile for Swift 5 and then perform all recommended migrations and fix deprecation warnings.

**Steps to test this PR**:

1. Ensure you're running Xcode 10.2 
1. Clear your Carthage caches - delete the Carthage folder in the project and follow instructions here https://stackoverflow.com/a/45504898/73479 
1. Update to latest Carthage (optional but recommended)
1. Check out this branch and execute `run_tests.sh`
1. Ensure you can run the app - do some exploratory testing 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/DetectedTracker.swift [hunk 1]
```diff
@@ -50,8 +50,11 @@ public struct DetectedTracker {
 
 extension DetectedTracker: Hashable {
 
-    public var hashValue: Int {
-        return "\(url) \(blocked) \(String(describing: networkName)) \(String(describing: category))".hashValue
+    public func hash( into hasher: inout Hasher) {
+        hasher.combine(url)
```
Comment (subsymbolic) [inline Core/DetectedTracker.swift:54]: Good change!

File: DuckDuckGo/TabViewControllerLongPressMenuExtension.swift [hunk 2]
```diff
@@ -66,7 +66,7 @@ extension TabViewController {
     
     private func onReadingAction(forUrl url: URL) {
         Pixel.fire(pixel: .longPressMenuReadingListItem)
-        try? SSReadingList.default()?.addItem(with: url, title: nil, previewText: nil)
+        ((try? SSReadingList.default()?.addItem(with: url, title: nil, previewText: nil)) as ()??)
```
Comment (subsymbolic) [inline DuckDuckGo/TabViewControllerLongPressMenuExtension.swift]: Optional: Is there a way to clean the syntax up here? It's not very readable. Not a blocker, if it doesn't work out I'll give it a try when I upgrade XCode too.

**Review by subsymbolic — COMMENTED** [review]
Just a comment from me. I'll leave the review approval with @bwaresiak

File: DuckDuckGo/TabViewControllerLongPressMenuExtension.swift (hunk 2 above)
Comment (brindy) [reply DuckDuckGo/TabViewControllerLongPressMenuExtension.swift]: Xcode made this change, but let me have a look.

Comment (brindy) [reply DuckDuckGo/TabViewControllerLongPressMenuExtension.swift]: I reverted it - doesn't see to be a problem.

**Review by bwaresiak — APPROVED** [review]
LGTM! 👍

**Comment by brindy** [comment]
Thanks!

**Pull request merged**